feature_engine/selection/drop_psi_features.py

```diff
@@ -387,28 +382,27 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None):
         if self.switch:
             test_df, basis_df = basis_df, test_df
 
-        if self.threshold == "auto":
-            threshold = self._calculate_auto_threshold(
-                basis_df.shape[0], test_df.shape[0]
-            )
-        else:
-            threshold = self.threshold
-
-        # set up the discretizer
-        if self.strategy == "equal_width":
-            bucketer = EqualWidthDiscretiser(bins=self.bins)
-        else:
-            bucketer = EqualFrequencyDiscretiser(q=self.bins)
+        # set up the discretizer for numerical variables
+        if len(self.num_variables_) > 0:
```

Comment:
Using PSI if there are only categorical features is perfectly fine and correct, so in my opinion, I would not raise an error or a warning in case there are no numerical values.

Reply:
Not sure about using PSI on categorical features is "perfectly fine". Chi2 is a better way of doing it.

Reply:
What's the reason for considering chi2 to be better than PSI?

Reply:
chi2 is very well studied and specifically designed to compare the frequency of categorical distributions (here more info: https://uk.sagepub.com/en-gb/eur/discovering-statistics-using-r/book236067). I think at some point I also read the chi2 article, but long forgotten.

I am also not sure I'd go and say "perfectly fine" to use PSI instead of chi2. Is this statement just based on that thesis? How well accepted is it in the community?

In my opinion, what happens is that if you make something available out of the box, like we are doing here, then less experienced users may be mislead to think that this is a proper way of assessing categorical variables. And I think, they should use chi2, based on my understanding of the consensus.

The thesis says that the PSI behaves like chi2, OK, it doesn't change the consensus the way I see it.

A follow up question could be, why does one use PSI instead of using other tests to compare continuous distributions, like the KolmogorovSmirnov test. I personally do not have the answer to that question, I haven't used the PSI much. So on that line, if we do discretize continuous variables to use the PSI, then why not apply it on categoricals as well, it seems like less work.

So in summary, what we need to decide is if we should we raise a warning if there are no numerical variables. And we should probably expand the user documentation with some discussion on chi2 vs PSI for categorical variables.

Reply:
I think my comment had been misinterpreted. I did not want to say that it is perfectly fine to use one metric over the other. What I said is that it is perfectly fine to use PSI to measure feature drift over a dataset of categorical variables only and thus it is not needed to raise a warning in this situation.

Reply:
Thank you!